### PR TITLE
Add pipeline events

### PR DESCRIPTION
**Pull request by thors**
Changes proposed in this pull request:
- Adds events for pipeline / workflow events by implementing org.jenkinsci.plugins.workflow.flow.GraphListener
- [x] Includes tests covering the new functionality? 
  -- src/main/site/examples/4_HandlePipelineEvents.groovy
- [x] Ready for review
- [x] Follows CONTRIBUTING rules
  -- Not sure which rules, but tried to follow implementation style for other events. New files can be used under MIT license (as stated in LICENSE file)

@nickgrealy

### CODE REVIEW

File: src/main/groovy/org/jenkinsci/plugins/globalEventsPlugin/GlobalRunListener.java [hunk 1]
```diff
@@ -34,6 +40,28 @@ public GlobalRunListener() {
...
+                    FlowExecution ex = ((WorkflowRun) run).getExecutionPromise().get();
+                    ex.addListener(new GlobalWorkflowListener(run));
+                    /*
+                    * Preferably use catch (InterruptedException | ExecutionException e),
+                    * but requires -source 1.7 flag.
+                    */
+                } catch (InterruptedException e) {
+                    e.printStackTrace(listener.getLogger());
+                    listener.error("Not able to get Workflow listener for this job");
+                } catch (ExecutionException e) {
```
Comment (Jimilian) [inline src/main/groovy/org/jenkinsci/plugins/globalEventsPlugin/GlobalRunListener.java]: This `ExecutionException` could be replaced by generic one - `Exception`. In this case you will be able to remove `InterruptedException` as well.

File: src/main/groovy/org/jenkinsci/plugins/globalEventsPlugin/GlobalRunListener.java [hunk 2]
```diff
@@ -52,13 +52,10 @@ public void run() {
                     * Preferably use catch (InterruptedException | ExecutionException e),
                     * but requires -source 1.7 flag.
                     */
-                } catch (InterruptedException e) {
+                } catch (Exception e) {
                     e.printStackTrace(listener.getLogger());
                     listener.error("Not able to get Workflow listener for this job");
-                } catch (ExecutionException e) {
```
Comment (Jimilian) [inline src/main/groovy/org/jenkinsci/plugins/globalEventsPlugin/GlobalRunListener.java]: You forgot to remove `import`

**Comment by nickgrealy** [comment]
How's this pull request coming along? Is it just waiting on the `import`?

**Comment by Jimilian** [comment]
Hi! @thors is on vacation now. If I remember correctly, removing `import` didn't fix issue. It looks like it needs dependency to `pipeline plugin`. What do you think? Is it ok? (Pipeline plugin should be installed by default since 2.*).

**Comment by nickgrealy** [comment]
Yeh, that should be fine. I changed the core jenkins plugin to 2.7.1 (in build.gradle - #23), so if 2+ has pipeline plugin by default, then no problem!

**Comment by afeblot** [comment]
Hi,
Should this PR be considered dead?
I'm looking for such functionality.

**Comment by Jimilian** [comment]
@afeblot, yes. more or less. Basically, Groovy Event Listener supports pipeline jobs. The problem is that each job will be considered as one big chunk :) What are you looking for?

**Comment by thors** [comment]
see last comment

**Pull request closed without merging**